### DISCUSSION TITLE
Media Keys not working in Windows

### DISCUSSION THREAD

**Discussion by DarthTerps** [audacious-media-player/audacious#1517] [General]
I have installed the 4.4.1 version on Windows 11.

The media keys do not work in Windows (If I boot to Linux, they work fine. Yes, I know)

I have reset the keyboard, installed the latest drivers, and the Media Keys do work in other Media Players (WinAmp, Apple Music, Spotify, etc.)

Any other help would be GREATLY appreciated. Volume/Mute keys work, but the Play/Pause, Next, Last, and Stop keys do not.

**Comment by radioactiveman** [comment]
Unfortunately there is no simple fix for this. The code to support media keys on Windows is simply not there. A separate Windows-only plugin for Audacious may be an option but someone needs to develop it first.

Maybe @jlindgren90 can give more insights. Qt supports such [keys](https://doc.qt.io/qt-6/qt.html#Key-enum) with `Qt::Key_Media***`.

**Reply by jlindgren90** [reply]
I don't really have anything to add, I don't know enough of the Windows API(s) to know what to use for this.

**Reply by radioactiveman** [reply]
I found https://github.com/Skycoder42/QHotkey/ which could be an option. Either by using this library directly or to find out how it works internally.

**Comment by richo67** [comment]
Maybe one possible step would be to use AutoHotKey. I am trying that, but I need to know WINDOW CLASS of the window which normaly receives the CTRL-ENTER and CTRL . for handling play and stop.

Can anybody help with it?
-r-

**Reply by ghost** [reply]
I wrote my own AutoHotkey script, but I used an alternative approach:  
I use `audtool.exe` to control the playback, rather than sending keyboard shortcuts to the `audacious.exe` window.

```autohotkey
#Requires AutoHotkey v2.0
#SingleInstance

#HotIf WinExist("ahk_exe audacious.exe")

Media_Prev::Run "C:\Program Files (x86)\Audacious\bin\audtool.exe --playlist-reverse",, "Hide"
Media_Next::Run "C:\Program Files (x86)\Audacious\bin\audtool.exe --playlist-advance",, "Hide"
Media_Stop::Run "C:\Program Files (x86)\Audacious\bin\audtool.exe --playback-stop",, "Hide"
Media_Play_Pause::Run "C:\Program Files (x86)\Audacious\bin\audtool.exe --playback-playpause",, "Hide"

#HotIf
```
The cool thing about it, is that it works equally well with the Qt, GTK, and “Winamp Classic” interfaces.
